differs/emerge_diff.go

```diff
@@ -0,0 +1,117 @@
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package differs
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	pkgutil "github.com/GoogleContainerTools/container-diff/pkg/util"
+	"github.com/GoogleContainerTools/container-diff/util"
+	"github.com/sirupsen/logrus"
+)
+
+//Emerge package database location
+const emergePkgFile string = "/var/db/pkg"
+
+type EmergeAnalyzer struct {
+}
+
+func (em EmergeAnalyzer) Name() string {
+	return "EmergeAnalyzer"
+}
+
+// EmergeDiff compares the packages installed by apt-get.
+func (em EmergeAnalyzer) Diff(image1, image2 pkgutil.Image) (util.Result, error) {
+	diff, err := singleVersionDiff(image1, image2, em)
+	return diff, err
+}
+
+func (em EmergeAnalyzer) Analyze(image pkgutil.Image) (util.Result, error) {
+	analysis, err := singleVersionAnalysis(image, em)
+	return analysis, err
+}
+
+func (a EmergeAnalyzer) getPackages(image pkgutil.Image) (map[string]util.PackageInfo, error) {
+	var path string
+	path = image.FSPath
+	switch path {
+	case "":
+		path = emergePkgFile
+	default:
+		path = filepath.Join(path, emergePkgFile)
+	}
+	packages := make(map[string]util.PackageInfo)
+	if _, err := os.Stat(path); err != nil {
+		// invalid image directory path
+		logrus.Errorf("Invalid image directory path %s", path)
+		return packages, err
+	}
+
+	contents, err := ioutil.ReadDir(path)
+	if err != nil {
+		logrus.Errorf("Non-content in image directory path %s", path)
+		return packages, err
+	}
+
+	for i := 0; i < len(contents); i++ {
+		c := contents[i]
+		pkgPrefix := c.Name()
+		pkgContents, err := ioutil.ReadDir(filepath.Join(path, pkgPrefix))
+		if err != nil {
+			return packages, err
+		}
+		for j := 0; j < len(pkgContents); j++ {
+			c := pkgContents[j]
+			pkgRawName := c.Name()
+			s := strings.Split(pkgRawName, "-")
+			if len(s) != 2 {
+				continue
+			}
+			pkgName, version := s[0], s[1]
+			pkgPath := filepath.Join(path, pkgPrefix, pkgRawName, "SIZE")
+			size := getPkgSize(pkgPath)
+			currPackage := util.PackageInfo{Version: version, Size: size}
+			fullPackageName := strings.Join([]string{pkgPrefix, pkgName}, "/")
+			packages[fullPackageName] = currPackage
+		}
+	}
+
+	return packages, nil
+}
+
+func getPkgSize(pkgPath string) int64 {
+	var sizeFile *os.File
+	var err error
+	sizeFile, err = os.Open(pkgPath)
+	if err != nil {
+		logrus.Debugf("unable to open SIZE for pkg %s", pkgPath)
+		return 0
+	}
+	defer sizeFile.Close()
+	fileBody, err := ioutil.ReadAll(sizeFile)
+	if err != nil {
+		logrus.Debugf("unable to read SIZE for pkg %s", pkgPath)
+		return 0
```

Comment:
Masking the error seems weird here. I think err should be returned rather than an inaccurate size given.

Reply:
make sense, I will fix this.